src/Factories/ConfigManagerFactory.php

```diff
@@ -0,0 +1,99 @@
+<?php declare(strict_types=1);
+
+namespace StrictPhp\HttpClients\Factories;
+
+use Generator;
+use StrictPhp\HttpClients\Clients\CacheResponse\Config as CacheResponseConfig;
+use StrictPhp\HttpClients\Clients\CustomizeRequest\Config as CustomizeRequestConfig;
+use StrictPhp\HttpClients\Clients\Retry\Config as RetryConfig;
+use StrictPhp\HttpClients\Clients\Sleep\Config as SleepConfig;
+use StrictPhp\HttpClients\Clients\Store\Config as StoreConfig;
+use StrictPhp\HttpClients\Contracts\ConfigContract;
+use StrictPhp\HttpClients\Managers\ConfigManager;
+
+/**
+ * @phpstan-type config array<class-string<ConfigContract>, array<string, mixed>|null>|array<ConfigContract>
+ */
+final class ConfigManagerFactory
+{
+    /**
+     * @var array<string, class-string<ConfigContract>>
+     */
+    private readonly array $configAliases;
+
+    /**
+     * @param array<string, class-string<ConfigContract>> $configAliases
+     */
+    public function __construct(
+        private readonly string $keyDefault = 'default',
+        array $configAliases = [],
+    ) {
+        $this->configAliases = $configAliases + [
+            'cacheResponse' => CacheResponseConfig::class,
+            'customizeRequest' => CustomizeRequestConfig::class,
+            'retry' => RetryConfig::class,
+            'sleep' => SleepConfig::class,
+            'store' => StoreConfig::class,
+        ];
+    }
+
+    /**
+     * config array represent
+     * [
+     *  host => [Config::class / alias => [constructor parameters]]
+     * ]
+     *
+     * [
+     *  'strictphp.com' => [
+     *     StrictPhp\HttpClients\Clients\CacheResponse\Config::class => [
+     *      'ttl' => null,
+     *      'saveOnly' => true,
+     *     ],
+     *     // or use alias
+     *     'cacheResponse' => [
+     *      'ttl' => null,
+     *      'saveOnly' => true,
+     *     ],
+     *     new StrictPhp\HttpClients\Clients\Sleep\Config(300, 1000),
+     *   ]
+     * ]
+     * @param array<string, config> $config
+     */
+    public function create(array $config): ConfigManager
+    {
+        $configManager = new ConfigManager();
+
+        if (isset($config[$this->keyDefault])) {
+            $configManager->addDefault($this->buildConfigManager($config[$this->keyDefault]));
+        }
+        unset($config[$this->keyDefault]);
```

Comment:
není rychlejší to ve foreachu ignorovat než unsetovat?

Reply:
myslíš mít v cyklu `if`?

unset je rychlejší jak pro velké tak pro malé pole, testoval jsem